code/game/objects/items/handcuffs.dm

```diff
@@ -70,51 +72,68 @@
 	fire = 50
 	acid = 50
 
-/obj/item/restraints/handcuffs/attack(mob/living/carbon/C, mob/living/user)
-	if(!istype(C))
+/obj/item/restraints/handcuffs/attack(mob/living/target_mob, mob/living/user)
+	if(!iscarbon(target_mob))
 		return
 
-	if(SEND_SIGNAL(C, COMSIG_CARBON_CUFF_ATTEMPTED, user) & COMSIG_CARBON_CUFF_PREVENT)
+	INVOKE_ASYNC(src, PROC_REF(attempt_to_cuff), target_mob, user) // avoid locking up the attack chain with sleeps
```

Comment:
You don't need to worry about locking up the attack chain with sleeps, in fact it's generally not a bad thing (to prevent `afterattack` from getting called instantly)

Reply:
alright